### ISSUE TITLE
data from input not populating variables in requests when clicking retry in collection runner after first run

### ISSUE THREAD

**Issue by dividendsolo** [a85/POSTMan-Chrome-Extension#4824]
Please read through the [guidelines](https://github.com/postmanlabs/postman-app-support#guidelines-for-reporting-issues) before creating a new issue.

**Describe the bug**
Clicking retry after a first run in collection runner with a data sheet(JSON in my case) is rerunning the requests, but no data is showing in the previously run requests.  

Example:
1st call with data sheet where 2 is set as 'number' for one of the iters:
_url-blahblah-endpoint?number=2_
is working as expected.

2nd call with same data sheet after clicking 'Retry' button:
_url-blahblah-endpoint?number=_  is getting sent as the request.
It's like the data is gone or not cached.

**To Reproduce**
1. Create a collection of GETS that have variables in the request, eg. {{number}} for above
1. Click 'Runner'
2. Select your collection
3. Select a data file, preferably JSON format
4. Select 'application/json' for Data File Type
5. Check Persist Variables checkbox
4. Click Run datasheets
7. Note the url being sent including your data you have set (i.e., what is the {{number} being sent?}
8. Click 'Retry'
9. See the variable value being sent in the url is gone ("")

**Expected behavior**
I would expect the 'Retry' would do just that, retry the same requests with data set in the first attempt.

**Screenshots**
Before first run:
![image](https://user-images.githubusercontent.com/38703245/42700777-bdd8262c-8692-11e8-8fbb-4c5250e9a056.png)

Results after first run, you can see data populated the urls (first call is empty on purpose):
![image](https://user-images.githubusercontent.com/38703245/42700918-26289950-8693-11e8-9967-f23db181353d.png)

Results after 'Retry', no data in variables:
![image](https://user-images.githubusercontent.com/38703245/42700971-4a8c0b9c-8693-11e8-93bd-a3ce6020a1cd.png)

**App information (please complete the following information):**
 - App Type Native App
 - Postman Version v6.1.4 
 - OS: Windows 10 Enterprise

**Additional context**
The workaround seems to be to click 'New' and run it again from scratch.

**Comment by sivcan** [comment]
@jameseg We do not store data file paths anywhere in the Postman Environment.
This will be solved in a similar way when this issue will be resolved: https://github.com/postmanlabs/postman-app-support/issues/4320

We'll be tracking further updates on that thread.

**Comment by dividendsolo** [comment]
?  The retry button doesn't work correctly. If you're not saving the data in the environment, then it's broken and therefore a bug  How can you expect the retry button to work if you have no access to the data file after it's run??

**Comment by sivcan** [comment]
@jameseg It's a bug but the only reason we are not storing any data file paths is because of security reasons associated with it. 

You chose the data file and the runner runs the run.
However, once the run is finished we discard all the file paths and just store the run's result.
On retrying the run, it would not take up any data file that was previously provided before running the run.

I understand it's a bug but the thread that I mentioned above aims to solve the use case of storing file paths throughout the app in a secure way.

We'll be tracking all the issues related to file paths in the same thread. :)

**Comment by dividendsolo** [comment]
Ok fair enough. I'm wondering why save the path though. Why not just create and store a temp data file off the original until re-import or the client is closed?

**Comment by sivcan** [comment]
Thanks for the suggestion. We'll keep a note of this. 
We'll update the main thread once we have an update regarding the data file paths. :)

**Comment by Maor-Katz** [comment]
WHY IS IT CLOSED???

**Comment by vdespa** [comment]
The way this currently works is very confusing and frustrating. 

@sivcan If a timely fix is not possible or too complicated, at least disable the "Retry" button after selecting a data file. Otherwise, it can be very confusing.

And why exactly was this issue closed?

**Comment by ghost** [comment]
@sivcan The issue was reported last year in July. The latest version 7.0.9 still has the issue. 
Can we know expected resolution date?

**Comment by davepile** [comment]
This is still an issue. It took me a good couple of hours to get here after checking my own prerequests and test scripts.
Disabling the Retry button from the Runner screen if there is a data file, or some sort of warning would very much welcomed.
